apps/judicial-system/web/src/routes/Prosecutor/CreateDetentionRequest/StepOne/StepOne.tsx

```diff
@@ -967,42 +944,29 @@ export const StepOne: React.FC = () => {
                     !workingCase.requestedCourtDate ||
                     Boolean(workingCase.courtDate)
                   }
-                  onBlur={(evt) => {
-                    const time = padTimeWithZero(evt.target.value)
-
-                    if (workingCase.requestedCourtDate) {
-                      const requestedCourtDateMinutes = parseTime(
-                        workingCase.requestedCourtDate,
-                        time,
-                      )
-                      const validateTimeEmpty = validate(time, 'empty')
-                      const validateTimeFormat = validate(time, 'time-format')
-
-                      setWorkingCase({
-                        ...workingCase,
-                        requestedCourtDate: requestedCourtDateMinutes,
-                      })
-
-                      if (
-                        validateTimeEmpty.isValid &&
-                        validateTimeFormat.isValid
-                      ) {
-                        updateCase(
-                          workingCase.id,
-                          parseString(
-                            'requestedCourtDate',
-                            requestedCourtDateMinutes,
-                          ),
-                        )
-                      } else {
-                        setRequestedCourtTimeErrorMessage(
-                          validateTimeEmpty.errorMessage ||
-                            validateTimeFormat.errorMessage,
-                        )
-                      }
-                    }
-                  }}
-                  onFocus={() => setRequestedCourtTimeErrorMessage('')}
+                  onChange={(evt) =>
+                    validateAndSetTime(
+                      'requestedCourtDate',
+                      workingCase.requestedCourtDate,
+                      evt.target.value,
+                      ['empty', 'time-format'],
+                      workingCase,
+                      setWorkingCase,
+                      requestedCourtTimeErrorMessage,
+                      setRequestedCourtTimeErrorMessage,
+                    )
+                  }
+                  onBlur={(evt) =>
+                    validateAndSendTimeToServer(
+                      'requestedCourtDate',
+                      workingCase.requestedCourtDate,
+                      evt.target.value,
+                      ['empty', 'time-format'],
+                      workingCase,
+                      updateCase,
+                      setRequestedCourtTimeErrorMessage,
+                    )
+                  }
```

Comment:
This file is a monster - over 1000 lines. Just saying.

Reply:
We are about to split the functionality up, so that should help with that.

Reply:
We should probably move the CreateCaseMutation into a separate mutation file as that's close to a 100 lines alone 😅